src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1426,28 +1421,32 @@ private void putAllCookiesInASingleHeader(HttpState state, HttpConnection conn,
 	 * https://issues.apache.org/jira/browse/HTTPCLIENT-1356
 	 *  
 	 * @param cookieheader
-	 * @return a list of name/value pair
+	 * @return a list of cookies corresponding to name of the cookie and its value, an empty list if empty or not a cookie line 
 	 */
-	private HeaderElement[] parseCookieHeaders(Header cookieheader) {
-		String cookies_string = cookieheader.toString();
-		cookies_string = cookies_string.substring("Cookie: ".length());
-		String[] cookies = cookies_string.split(";");
-		List<HeaderElement> headerElements = new ArrayList<HeaderElement>();
-		for (String cookie : cookies){
-			String[] parts = cookie.split("=");
-			HeaderElement headerElement = new HeaderElement(parts[0].trim(),parts[1].trim());
-			headerElements.add(headerElement);
+	public static List<Cookie> parseCookieHeader(String host, String cookieHeaderValue) {
```

Comment:
It could be just default accessible (I don't think other "external" classes need to call this method).